ts/screens/wallet/payment/ManualDataInsertionScreen.tsx

```diff
@@ -236,17 +254,21 @@ class ManualDataInsertionScreen extends React.Component<Props, State> {
   }
 }
 
-const mapDispatchToProps = (dispatch: Dispatch) => ({
+const mapDispatchToProps = (dispatch: Dispatch, props: OwnProps) => ({
   navigateToWalletHome: () => dispatch(navigateToWalletHome()),
   navigateToTransactionSummary: (
     rptId: RptId,
     initialAmount: AmountInEuroCents
   ) => {
+    if (props.navigation.getParam("isInvalidAmount")) {
+      props.navigation.setParams({ isInvalidAmount: false });
```

Comment:
Why you need to set the param here?

Reply:
mapStateToProps must be used to get stuff from state and create props from the stuff.

Reply:
Fixed in the commit (9ca441a). I have used the willFocus event.

Reply:
@fpersico I did another test on a physical android device after the change (screencast below). I confirm that the amount is reset before the screen is visible.

![retry_nexus](https://user-images.githubusercontent.com/10615049/59492786-fb57a680-8e89-11e9-9cc2-2fcbe8f75109.gif)